### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by pksohn**
This PR includes many changes to make UrbanSim compatible with both Python 2.7 and 3.X. Changes include (in order of importance):

* Added future import of true division and forced floor division in a few places where it seemed necessary, mostly `mnl.py` and `pmat.py`. @fscottfoti it would be great if you wouldn't mind taking a look at these (I added TODOs in the important spots).
* Changes to iterators like `dict.iteritems()` and `itertools.izip()` which were removed in Python 3. I replaced iteritems with `dict.items()` which is inefficient in Python 2 since it doesn't use a generator, but most use cases are very simple so it shouldn't make a different in performance.
* Convert implicit relative imports to explicit relative imports where appropriate.
* Error message strings in the `e.message` format converted to `str(e)`.
* Updated imports to reflect changes to standard library.
* Some tests for functions that use Python sets were failing due to reordered items; tests were changed to sort results before assertions.
* Some tests were failing in Windows due to int64/int32 mismatch; dtype checking was disabled for these tests.
* Travis CI and Appveyor CI updated to build in both Python 2.7 and 3.5. We'll need to update these once Orca 1.4 is released and sanfran_urbansim's equivalent branch is merged.
* PEP 8 formatting changes (the `pep8` package was dropped in CI builds in favor of the new `pycodestyle`).
* Updated print statements.

I'll leave this PR open for a bit. It'd be great if folks are able to test this on Python 3.X and make sure we aren't missing important edge cases or compatibility issues.

### CODE REVIEW

**Comment by janowicz** [comment]
Thanks for spearheading the python3 transition @pksohn!   The changes look good to me.  Tried a few different urbansim workflows, and it all seemed to work as expected, and I'll continue using this branch for testing. 

I am not an experienced user of python3, so like you mentioned it'll be great to get other reviewers/testers from the community too.

File: urbansim/urbanchoice/mnl.py [hunk 1]
```diff
@@ -33,7 +33,8 @@ def mnl_probs(data, beta, numalts):
     utilities = beta.multiply(data)
     if numalts == 0:
         raise Exception("Number of alternatives is zero")
-    utilities.reshape(numalts, utilities.size() / numalts)
+    # TODO confirm we want floor division here
+    utilities.reshape(numalts, utilities.size() // numalts)
```
Comment (fscottfoti) [inline urbansim/urbanchoice/mnl.py]: yes - it should be an int

File: urbansim/urbanchoice/mnl.py [hunk 2]
```diff
@@ -67,7 +68,8 @@ def mnl_loglik(beta, data, chosen, numalts, weights=None, lcgrad=False,
                stderr=0):
     logger.debug('start: calculate MNL log-likelihood')
     numvars = beta.size
-    numobs = data.size() / numvars / numalts
+    # TODO confirm we want floor division here
+    numobs = data.size() // numvars // numalts
```
Comment (fscottfoti) [inline urbansim/urbanchoice/mnl.py]: yes

File: urbansim/urbanchoice/mnl.py [hunk 3]
```diff
@@ -165,7 +167,8 @@ def mnl_simulate(data, coeff, numalts, GPU=False, returnprobs=True):
         probs = PMAT(probs.get_mat())
 
     probs = probs.cumsum(axis=0)
-    r = pmat.random(probs.size() / numalts)
+    # TODO confirm we want floor division here
+    r = pmat.random(probs.size() // numalts)
```
Comment (fscottfoti) [inline urbansim/urbanchoice/mnl.py]: yup

File: urbansim/urbanchoice/mnl.py [hunk 4]
```diff
@@ -219,7 +222,8 @@ def mnl_estimate(data, chosen, numalts, GPU=False, coeffrange=(-3, 3),
     atype = 'numpy' if not GPU else 'cuda'
 
     numvars = data.shape[1]
-    numobs = data.shape[0] / numalts
+    # TODO confirm we want floor division here
+    numobs = data.shape[0] // numalts
```
Comment (fscottfoti) [inline urbansim/urbanchoice/mnl.py]: yes

File: urbansim/urbanchoice/pmat.py [hunk 5]
```diff
@@ -146,6 +150,7 @@ def subtract(self, mat, inplace=False):
 
     def divide_by_row(self, rowvec, inplace=False):
         if self.typ == 'numpy':
+            # TODO confirm we want true division here
```
Comment (fscottfoti) [inline urbansim/urbanchoice/pmat.py]: yes on all of these as well

**Comment by fscottfoti** [comment]
This all looks good to me too.  I'm sad to see python2.x go though - who wants to use parenthesis when printing!  It had a good run though - I've used python2.x for 16 years now.
